### ISSUE TITLE
No data to consider

### ISSUE THREAD

**Issue by acubino** [ggurov/hx711#1]
Hi.

I just connected the hx711 model to the P5 header but are unable to read any data.
I just try with 2 different sensor from 2 different manufacturer, but all time say "no data to consider".

Also I compile que hx711.c with different pinout:
# define CLOCK_PIN       30
# define DATA_PIN        28

But the response is the same all time.

I attach the picture of my raspberry and the sensor
![bascula](https://cloud.githubusercontent.com/assets/6959449/3720310/a504c0ec-1652-11e4-93b4-706cf70a33ee.jpg)

Any help?

**Comment by rcrum003** [comment]
How do you wire the load cell to the ADS1115  
I have Power from PI to VDD, GND to GND, SCL and SDA all from the PI connected to corresponding pins on ADS1115.

The load cell has 4 wires Red, Black, White, Green - and the ADS1115 has A0, A1, A2, A3, ADDR, ALRT all available.

Do i power the load cell with the VDD/GND ports?

Any advice would be appreciated.

**Comment by ggurov** [comment]
yes, you power the load cell from the ads , black and red are power, black
is ground, red is 5v. other two you hook up to a0 and a1, order doesn't
matter.

the example for ads1115 has a differential mode script, use that but turn
the resolution to .256v(the gain setting ), then you will get useful data.

you will need to calibrate the scale with known weights.  you can use
volumes of water or proper weights,  unless you are just looking for rate
of change.

-gurov

How do you wire the load cell to the ADS1115

I have Power from PI to VDD, GND to GND, SCL and SDA all from the PI
connected to corresponding pins on ADS1115.

The load cell has 4 wires Red, Black, White, Green - and the ADS1115 has
A0, A1, A2, A3, ADDR, ALRT all available.

Do i power the load cell with the VDD/GND ports?

Any advice would be appreciated.

—
Reply to this email directly or view it on GitHub
<https://github.com/ggurov/hx711/issues/1#issuecomment-146044959>.

**Comment by JoeHogan** [comment]
Trying to revive this conversation - I have a Tessel 2 which can communicate via I2C. When trying to read differential data from the ADS1115 across pins 0 and 1 on the ADS1115, i always get returned: Buffer 85 83. I have it configured with the default address (by connecting the address pin to ground) and I set the gain to the highest. Below is my code (tessel is JS based). Any thoughts as to what im doing wrong?

```
var tesselPort = tessel.port.B;
var slaveAddress = 0x48; //binary is 01001000 with ADDR pin to ground
var i2c = new tesselPort.I2C(slaveAddress); //create i2c port at this address
i2c.send(new Buffer([0x90, 0x01, 0x8E, 0x83]), function (err) { //set config register with gain to highest 0x8E (10001110)
    if(err) console.log(err);
    console.log('configuration complete');
    i2c.send(new Buffer([0x90, 0x00]), function (err) {
        if(err) console.log(err);
        console.log('pointed to conversion register in continuous mode');
        setInterval(function(){ //continuously read from conversion register every second
            console.log('querying...'); 
            i2c.transfer(new Buffer([0x91]), 2, function (err, rx) { //second and third bytes should be the data returned (the 2 represents the expected two bytes returned)
              if(err) console.log('ERR: '+err)
              console.log('buffer returned by I2C slave', rx); //always getting <Buffer 85 83>
            });
        },1000);
    });
});
```

**Comment by ggurov** [comment]
not sure about that platform. the reason for picking ads1115 was the
existence of the libraries that are easy to get started with.

did you ever get any response from the ads in straight(not differential)
ADC mode ?

maybe start with using pure SMBUS python module (on rpi or another bench
device) to signal the chip and then translate code to js framework you've
got.

hard to tell from the code, I don't know the init procedure for the chip
the library takes care of.

**Comment by JoeHogan** [comment]
Thanks so much for the response... Maybe my problem is that i cant tell from the datasheet how to set to differential/non-differential. It says Bit 8 controls Continuous vs Straight-Shot, and ive tried both of those with the same results. The datasheet says Bits 14-12 control the multiplexer, which is where i assume you can set differential inputs - here im using the default: 000 : AINP = AIN0 and AINN = AIN1 (default) - and just plugging my inputs from the strain gauge into inputs 0 and 1 on the ADS1115 - but im not sure if there is a separate setting to set the ADS1115 to differential mode

**Comment by ggurov** [comment]
the ads1115 Python module code is not very complicated, bunch of static
names and smbus calls, I've modified it once to run a $3 24 bit ADC from
the datasheet. take a look it.

**Comment by xhr0428** [comment]
I was using HX711 then saw this thread and decided to try ADS1115.
I wired
          3.3V - - - - - - - - - - VDD - - - - - - - - - - Red
         GND - - - - - - - - - - GND - - - - - - - - - - Black
          SCL - - - - - - - - - - SCL
          SDA - - - - - - - - - - SDA 
                                         ADDR-GND
                                         A0    - - - - - - - - - -  White
                                         A1    - - - - - - - - - -  Green
        Arduino Nano       ADS115                     Load Cell

Gain was set to 16 the highest.
The differential reading I got was very small (0-83). I had to pull the load cell really hard to get 83.
Did I miss anything?
